src/arch/xtensa/include/arch/init.h

```diff
@@ -46,9 +46,14 @@
  */
 static inline void exception(void)
 {
+	uintptr_t epc1;
+
+	__asm__ __volatile__("rsr %0, EPC1" : "=a" (epc1) : : "memory");
+
 	/* now panic and rewind 8 stack frames. */
 	/* TODO: we could invoke a GDB stub here */
-	panic_rewind(SOF_IPC_PANIC_EXCEPTION, 8 * sizeof(uint32_t), NULL);
+	panic_rewind(SOF_IPC_PANIC_EXCEPTION, 8 * sizeof(uint32_t),
+		     NULL, &epc1);
```

Comment:
No need to do a redirection and pass ptr here (suggests that it is "inout", we should use more const btw). It is already a ptr, so passing 0 (=NULL) in panic (non-exception) case is OK.

Reply:
Unfortunately it is the only working version. I have started with a standard passing by value but it result in a strange behaviour like changing the value when in panic_rewind().